client/lib/media/utils/is-transient-media-id.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * External dependencies
+ */
+import { startsWith } from 'lodash';
+
+/**
+ * Check whether a given string is a transient media ID. Transient media IDs
+ * are strings and start with 'media-'. Persisted media IDs are numbers.
+ *
+ * @param {string} id a media id we want to check
+ * @returns {boolean} true if given id is a transient media id, false otherwise
+ */
+function isTransientMediaId( id ) {
+	return typeof id === 'string' && startsWith( id, 'media-' );
```

Comment:
Maybe we could use `id.startsWith()` instead of the lodash alternative? Or is there a good reason to prefer the lodash version?

Reply:
good point @tyxla, I opted for the lodash method because the native `startsWith` [doesn't seem to be supported by IE11](https://caniuse.com/mdn-javascript_builtins_string_startswith). However, it may be polyfilled somehow?

Reply:
Yes, last time I checked, we were polyfilling `String.startsWith` already, so it should be safe to use.